src/google-maps/map-kml-layer/map-kml-layer.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {
+  Directive,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+import {MapEventManager} from '../map-event-manager';
+
+/**
+ * Angular component that renders a Google Maps KML Layer via the Google Maps JavaScript API.
+ *
+ * See developers.google.com/maps/documentation/javascript/reference/kml#KmlLayer
+ */
+@Directive({
+  selector: 'map-kml-layer',
+})
+export class MapKmlLayer implements OnInit, OnDestroy {
+  private _eventManager = new MapEventManager(this._ngZone);
+  private readonly _options = new BehaviorSubject<google.maps.KmlLayerOptions>({});
+  private readonly _url = new BehaviorSubject<string>('');
```

Comment:
This will pass an empty string to the map if we subscribe before the `url` has been assigned. Would that result in an error  from the Maps API? We could use a `ReplaySubject` here instead.

Reply:
Url is also in options, so this allows the user to specify url as its own property, or to just include it as an option. Url is also allowed to be an empty string or undefined by the api, so it should not cause errors if the user uses an empty string on purpose.